### PR TITLE
ref(cmd): clarify deis registry usage

### PR DESCRIPTION
**Pull request by bacongobbler**
related to deis/workflow#239. Also increases test coverage around this so a user cannot accidentally fat-finger an invalid key.

### CODE REVIEW

File: cmd/registry.go [hunk 1]
```diff
@@ -124,9 +124,13 @@ func parseInfo(item string) (string, string, error) {
 	parts := strings.Split(item, "=")
 
 	if len(parts) != 2 {
-		return "", "", fmt.Errorf(`%s is invalid, Must be in format key=value
+		return "", "", fmt.Errorf(`%s is invalid. Must be in format key=value
 Examples: username=bob password=s3cur3pw1`, item)
 	}
 
+	if parts[0] != "username" && parts[0] != "password" {
```
Comment (helgi) [inline cmd/registry.go]: We'll just have to remember to update this when `hostname` (will be needed for Builder in the future) and potentially other items become available

File: parser/registry.go [hunk 2]
```diff
@@ -61,15 +61,17 @@ Options:
 
 func registrySet(argv []string) error {
 	usage := `
-Sets registry information for an application.
-
-key/value pairs used to configure / authenticate against external registries
+Sets registry information for an application. These credentials are the same as you'd use when
```
Comment (mboersma) [inline parser/registry.go]: Could you rephrase the second sentence not to use "you"? Maybe:
> These credentials are the same as those used for `docker login` to the external registry.

Comment (bacongobbler) [inline parser/registry.go]: always with the first person, eh? Old habits die hard. Good catch.
